src/components/elements/text/text.tsx

```diff
@@ -95,6 +96,7 @@ export const Text = (props: IGenericElementProps<ITextSchema>) => {
 				ref={elementRef}
 				maxLines={!expanded ? maxLines : undefined}
 				data-testid={getTestId(id)}
+				weight={isNumber(weight) ? WEIGHT_MAPPING[weight] : weight}
```

Comment:
there's an additional logic that I overlooked here required for backward compatibility:
- if `weight` is number, perform `WEIGHT_MAPPING[weight]`
- else if `weight` is defined, use `weight`
- else if this is using V2 Text, fallback to the default weight based on the [second table in the migration docs](https://github.com/LifeSG/react-design-system/wiki/Migrating-to-V3#text)